src/Api/ApiServiceProvider.php

```diff
@@ -96,10 +96,14 @@ public function boot()
      */
     protected function registerNotificationSerializers()
     {
+        $this->app->singleton('flarum.api.notification_serializers', function () {
+            return [
+                'discussionRenamed' => BasicDiscussionSerializer::class
+            ];
+        });
+
         $blueprints = [];
-        $serializers = [
-            'discussionRenamed' => BasicDiscussionSerializer::class
-        ];
+        $serializers = $this->app->make('flarum.api.notification_serializers');
```

Comment:
Similarly, if we move this to boot it'll only evaluate the singleton after extensions have been processed.

Reply:
Taking a second look, the logic is already technically in boot, which is why it worked before. We should change it so that the singleton is registered in `register` and evaluated in `boot`.